controllers/deployment.go

```diff
@@ -133,6 +168,37 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, application *skipe
 			}
 
 			volumeMounts[i+1] = corev1.VolumeMount{Name: volumes[i+1].Name, MountPath: file.MountPath}
+			VolNumber = i
```

Comment:
Is this correct? Does it actually work with `filesFrom` defined? Can we write a test case that checks for the correctness of `gcp` when combined with `filesFrom`?